beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -63,11 +63,28 @@ func (vs *Server) setExecutionData(ctx context.Context, blk interfaces.SignedBea
 					return errors.Wrap(err, "failed to get local payload value")
 				}
 
+				// Compare withdrawal roots. Default to local if they don't match.
+				r1, err := localPayload.WithdrawalsRoot()
+				if err != nil {
+					return errors.Wrap(err, "failed to get local withdrawals root")
+				}
+				r2, err := builderPayload.WithdrawalsRoot()
+				if err != nil {
+					return errors.Wrap(err, "failed to get builder withdrawals root")
+				}
+				matchWithdrawalRoot := bytes.Equal(r1, r2)
+				if !matchWithdrawalRoot {
+					log.WithFields(logrus.Fields{
+						"local":  r1,
+						"remote": r2,
+					}).Warn("Proposer: withdrawal roots don't match, using local")
+				}
```

Comment:
An alternative is to add here an `} else` and prevent continuing if we already know that we are not going to use the builder cause the withdrawals root doesn't match